Handle paging when getting commits

eparis opened the pull request:
The change after LGTM hook has been trigger incorrectly. See:

https://github.com/kubernetes/kubernetes/pull/13146
https://github.com/kubernetes/kubernetes/pull/15994

This handles paging on commits (which we shouldn't have more than 100
commits, but that is a code path this munger uses) and adds a touch more
debug info.

eparis committed:
Handle paging when getting commits

The change after LGTM hook has been trigger incorrectly. See:

https://github.com/kubernetes/kubernetes/pull/13146
https://github.com/kubernetes/kubernetes/pull/15994

This handles paging on commits (which we shouldn't have more than 100
commits, but that is a code path this munger uses) and adds a touch more
debug info.

lavalamp commented on mungegithub/github/github.go:
```diff
@@ -637,12 +634,14 @@ func (obj *MungeObject) GetStatus(requiredContexts []string) (string, error) {
 		return "failure", nil
 	}
+	// TODO If we have more than 100 statuses we need to deal with paging.
 	combinedStatus, response, err := config.client.Repositories.GetCombinedStatus(config.Org, config.Project, *pr.Head.SHA, &github.ListOptions{})
```
Is setting PerPage to 100 needed or useful?

lavalamp commented on mungegithub/mungers/lgtm_after_commit.go:
```diff
@@ -62,6 +62,7 @@ func (LGTMAfterCommitMunger) Munge(obj *github.MungeObject) {
 	}
 
 	if lastModified.After(*lgtmTime) {
+		glog.Infof("lgtm:%s  lastModified:%s", lgtmTime.String(), lastModified.String())
```
Print the PR #, or the logs will be confusing.

lavalamp commented:
Feel free to self-lgtm once you fix nits :)

eparis committed:
Update github test to handle raciness

It was possible that our test, which expected to see that our pr was
queued, already finished testing. So really we want to check for either.

lavalamp commented:
Did those commits actually break the travis tests? :)

eparis commented:
no, -tip is broken, not the tests.

lavalamp commented:
gotcha.

On Wed, Nov 11, 2015 at 10:51 AM, Eric Paris <notifications@github.com>
wrote:

> no, -tip is broken, not the tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/contrib/pull/213#issuecomment-155875437>.
>
